README.md

```diff
@@ -1201,6 +1204,34 @@ Zenstruck\Foundry\Test\TestState::addGlobalState(function () {
 only loaded once.
 2. The [`ResetDatabase`](#enable-foundry-in-your-testcase) trait is required when using global state.
 
+### PHPUnit Data Providers
+
+It is possible to use factories in
+[PHPUnit data providers](https://phpunit.readthedocs.io/en/9.3/writing-tests-for-phpunit.html#data-providers):
+
+```php
+use App\Factory\PostFactory;
+
+/**
+ * @dataProvider postDataProvider
+ */
+public function test_post_via_data_provider(PostFactory $factory): void
+{
+    $post = $factory->create();
+
+    // ...
+}
+
+public static function postDataProvider(): iterable
+{
+    yield [PostFactory::new()];
+    yield [PostFactory::new()->published()];
+}
+```
+
+**NOTE**: Due to data providers being computed early in the phpunit process, it is not possible to use
+[Factory Services](#factories-as-services) in them.
```

Comment:
I think if I'm reading this, I'm thinking:

> Ok. My service *is* a factory... so what should I do?

And... what *would* happen? Does it fallback to just trying to instantiate manually and that would work if you had optional args?

It might also be important to point out that you should be returning *factory* instances from your data providers (exactly like you've shown here) and not actually calling `->create()` until the real method, because `->create()` in the data provider wouldn't work because it's too early (would it error? Or just create but not persist?)